From 435198f33b56d7b875a8173a0227ddf0de285aa1 Mon Sep 17 00:00:00 2001
From: James Zhu <jzhums@gmail.com>
Date: Fri, 10 Aug 2018 00:31:39 +0800
Subject: [PATCH] drm/amdgpu: update tmr mc address
References: bsc#1100132
Patch-mainline: v4.18-rc7
Git-commit: 435198f33b56d7b875a8173a0227ddf0de285aa1

Update tmr mc address with firmware loading address
which is returned from PSP firmware

Backporting notes:

  * use function-local variable 'cmd_buf_mem'

Signed-off-by: James Zhu <James.Zhu@amd.com>
Reviewed-by: Alex Deucher <alexander.deucher@amd.com>
Acked-by: Huang Rui <ray.huang@amd.com>
Reviewed-by: Likun Gao <Likun.Gao@amd.com>
Signed-off-by: Likun Gao <Likun.Gao@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Cc: stable@vger.kernel.org
Acked-by: Thomas Zimmermann <tzimmermann@suse.de>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
@@ -133,6 +133,11 @@ psp_cmd_submit_buf(struct psp_context *p
 		msleep(1);
 	}

+	if (ucode) {
+		ucode->tmr_mc_addr_lo = cmd_buf_mem->resp.fw_addr_lo;
+		ucode->tmr_mc_addr_hi = cmd_buf_mem->resp.fw_addr_hi;
+	}
+
 	amdgpu_bo_free_kernel(&cmd_buf_bo,
 			      &cmd_buf_mc_addr,
 			      (void **)&cmd_buf_mem);
